src/api.jl

```diff
@@ -41,3 +41,14 @@ function surrogate(x, method::Surrogate)
     sg = surrogenerator(x, method)
     sg()
 end
+
+"""
+    surrogate(x,t, method::Surrogate) → s
+Create a  single timeseries `s` from the irregular time series `x` with time steps `t`
+based on the given `method`.
+If you want to generate more than one surrogate from `x`, you should use a [`surrogenerator`](@ref)
+"""
+function surrogate(x, t, method::Surrogate)

```

Comment:
`t` could be simply made an argument of the surrogate method, instead of expanding the API. This adds more complexity and more methods to keep track off, but I don't immediately see a direct benefit of having `t` here instead of an argument to e.g. the `LS` method.

Reply:
> ... I don't immediately see a direct benefit of having t here instead of an argument to e.g. the LS method.

@felixcremer I guess the intention is to plan the lombscargle periodogram for one set of time indices, the re-use those time indices for every surrogate? If so, as @Datseris suggests, it would make more sense to have `t` as an argument to the surrogate method. In the `surrogenerator` method (should be added, as for the other methods), you then plan once and re-use that plan for every surrogate. 

If `t` changes, then you add a new surrogate method (because you're no longer working with the same data set when `t` changes, so the user should create a new surrogate method instance).

Reply:
I made the time axis a field of the LS struct and implemented the surrogenerator functionality.